src/hooks/useContactsList.js

```diff
@@ -0,0 +1,115 @@
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query';
+import {
+  getStringNoLocale,
+  getThing,
+  removeThing,
+  getThingAll,
+  getUrl,
+  buildThing,
+  createThing,
+  setThing,
+  createSolidDataset,
+  getSolidDataset,
+  saveSolidDatasetAt
+} from '@inrupt/solid-client';
+import { RDF_PREDICATES } from '@constants';
+import useSession from './useSession';
+
+const makeContactIntoThing = ({ givenName, familyName, webId }) =>
+  buildThing(createThing({ name: encodeURIComponent(webId) }))
+    .addStringNoLocale(RDF_PREDICATES.Person, `${givenName} ${familyName}`)
+    .addStringNoLocale(RDF_PREDICATES.givenName, givenName)
+    .addStringNoLocale(RDF_PREDICATES.familyName, familyName)
+    .addUrl(RDF_PREDICATES.identifier, webId)
+    .addUrl(RDF_PREDICATES.URL, webId.split('profile')[0])
+    .build();
+
+const parseContacts = (data) => {
+  const contactThings = getThingAll(data);
+  const contacts = [];
+  contactThings.forEach((thing) => {
+    const contact = {};
+
+    contact.webId = getUrl(thing, RDF_PREDICATES.identifier);
+    contact.podUrl = getUrl(thing, RDF_PREDICATES.URL);
+    contact.givenName = getStringNoLocale(thing, RDF_PREDICATES.givenName);
+    contact.familyName = getStringNoLocale(thing, RDF_PREDICATES.familyName);
+    contact.person = getStringNoLocale(thing, RDF_PREDICATES.Person);
+    contacts.push(contact);
+  });
+  return contacts;
+};
+
+/**
+ *
```

Comment:
Could always flush out the JSDoc out later, hook seems to be working as intended.

Reply:
fixed